worker/sessionpool.go

```diff
@@ -18,7 +19,12 @@ func (s *Session) appendSector(ctx context.Context, sector *[rhpv2.SectorSize]by
 	}
 	storageDuration := uint64(s.Revision().Revision.WindowStart) - currentHeight
 	price, collateral := rhpv2.RPCAppendCost(s.settings, storageDuration)
-	return s.Append(ctx, sector, price, collateral)
+	root, err := s.Append(ctx, sector, price, collateral)
+	if err != nil {
+		return root, err
+	}
+	RecordContractSpending(ctx, s.revision.ID(), api.ContractSpending{Uploads: price})
```

Comment:
did you consider pushing it down into `func (s *Session) Write`? that is also where we record rpc metrics